stacks/api/function/enqueuer/src/http.ts

```diff
@@ -0,0 +1,150 @@
+import {Middlewares} from "@spica-server/core";
+import {EventQueue, HttpQueue} from "@spica-server/function/queue";
+import {Event, Http} from "@spica-server/function/queue/proto";
+import {Description, Enqueuer} from "./enqueuer";
+import express = require("express");
+import read = require("body-parser/lib/read");
+import bodyParser = require("body-parser");
+
+export class HttpEnqueuer extends Enqueuer<HttpOptions> {
+  description: Description = {
+    icon: "http",
+    name: "http",
+    title: "Http",
+    description: "Designed for APIs and Http Streaming"
+  };
+
+  private router = express.Router({mergeParams: true});
+
+  constructor(private queue: EventQueue, private http: HttpQueue, httpServer: express.Application) {
+    super();
+    this.router.use(
+      bodyParser.raw({
+        limit: "10mb",
+        type: "*/*"
+      })
+    );
+    this.router.use(this.handleUnhandled);
+    const stack = httpServer._router.stack;
+    httpServer.use("/fn-execute", this.router);
+    const expressInitIndex = stack.findIndex(l => l.name === "expressInit");
+    const layer = stack.splice(stack.length - 1, 1)[0];
+    stack.splice(expressInitIndex + 1, 0, layer);
+  }
+
+  private handleUnhandled(req, res) {
+    // By default express sends a default response if the OPTIONS route is unhandled
+    // https://github.com/expressjs/express/blob/3ed5090ca91f6a387e66370d57ead94d886275e1/lib/router/index.js#L640
+    if (!req.route) {
+      res.status(404).send({
+        message: "Invalid route",
+        url: req.originalUrl,
+        method: req.method,
+        engine: "Function"
+      });
+    }
+  }
+
+  private reorderUnhandledHandle() {
+    this.router.stack.splice(this.router.stack.findIndex(l => l.handle == this.handleUnhandled), 1);
+    this.router.use(this.handleUnhandled);
+  }
+
+  subscribe(target: Event.Target, options: HttpOptions): void {
+    const method = options.method.toLowerCase();
+    const path = options.path.replace(/^\/?(.*?)\/?$/, "/$1");
+
+    if (
+      options.preflight &&
+      options.method != HttpMethod.Get &&
+      options.method != HttpMethod.Head
```

Comment:
@thesayyn Does the condition suppose to be "if method is not get or head"?

Reply:
The preflight option is meaningless if it used with Get or Head method. So this check works like it is supposed to

Reply:
But the check uses AND. Shouldn't it be OR?

Reply:
Nope, this is the correct check to accomplish the desired behavior.

It should execute this "if block" if "options.preflight == true" AND "options.method  != get" AND "options.method != head"